File: stack & queue/QueueLL.java

public class QueueLL {

    static class Node {
        int key;
        Node next;
        Node(int key){
            this.key = key;
            next = null;
        }
    }

    Node front = null;
    Node rear = null;

    void enqueue(int key){
        Node node = new Node(key);
        if (rear == null){
            front = rear = node;
            return;
        }
        rear.next = node;
        rear = node;
    }

    void dequeue(){
        if (front == null)
            return;
        Node node = front;
        front = node.next;
        if (front == null)
            rear = null;
    }

    void print(){
        Node node = front;
        while (node!=null){
            System.out.print(node.key+" ");
            node = node.next;
        }
    }

    public static void main(String[] args) {
        
    }
}

File: queue/QueueLL.js
import Node from '../linked-list/Node.js';

class QueueLL {
    constructor() {
        this.front = null;
        this.rear = null;
    }

    isEmpty() {
        return this.rear === null;
    }

    peek() {
        if (this.front === null) {
            return -1;
        }
        return this.front.key;
    }

    enqueue(key) {
        let node = new Node(key);
        if (this.isEmpty()) {
            this.front = this.rear = node;
        } else {
            this.rear.next = node;
            this.rear = node;
        }
    }

    dequeue() {
        if (this.front === null) {
            return;
        }
        let key = this.front.key;
        this.front = this.front.next;
        if (this.front === null) {
            this.rear = null;
        }
        return key;
    }

    length() {
        if (this.front === null) {
            return -1;
        }
        let front = this.front;
        let length = 0;
        while (front !== null) {
            ++length;
            front = front.next;
        }
        return length;
    }

}

export default QueueLL;

/*  Usage
    const queue = new QueueLL();
    queue.enqueue(3)
    queue.enqueue(4)
    queue.enqueue(5)
    console.log(queue.peek())
    console.log(queue.dequeue())
    console.log(queue.peek())
*/